index.js

```diff
@@ -21,33 +18,25 @@ fs.mkdir(pathName, { recursive: true }, (error) => {
 
 fs.readFile(file, 'utf8', (err, data) => {
     if (err) {
-        return console.log(err)
+        throw err;
     }
 
     // the data is split into a new array called clippings
     const clippings = data.split('==========');
 
     // declare an empty array that will hold the books & highlights
-    let books = [];
-
-    // searchBooks searches in 'books' array & returns index of book if it exists
-    function searchBooks(books, title) {
-        for(let i = 0; i < books.length; i++) {
-            if (books[i].title === title) {
-                bookIndex = i;
-                return;
-            }
-        }
-        bookIndex = -1;
-        return bookIndex;
-    }
+    const books = {};
```

Comment:
after looking at what this program was doing (it errored out for me but it was easy to read and see what the goal was), I think your data structure was not ideal.
you had:
```
[{ title, highlight }, ...]
```
so to find highlights, or append them, you need to loop over your list of books. But crucially you don't really care about the order of the titles (even if you did this is probably better), so since the operations you want to do involve searching for a book by its title, better to use a datastructure that's conducive to searching by things by title/label/key.

Objects are PERFECT for this. By using an object we can eliminate a lot of the helper functions too!